src/app/participant-list/participant-list.component.ts

```diff
@@ -1744,23 +1744,45 @@ export class ParticipantListComponent implements OnInit {
     if (column && column.participantColumn) {
       name = column.participantColumn.name;
     }
-    if (personData && personData.data && name) {
-      let currentKey = Object.keys(personData.data).find(key => key === name);
-      let field = personData.data[currentKey];
-      if (field) {
-        if (column.options && column.options[0] && column.options[0].name) {
-          let fieldToShow = null;
-          if (column.additionalType === Filter.ACTIVITY_STAFF_TYPE) {
-            fieldToShow = column.options.find(nameValue => nameValue.name === field);
-          } else {
-            fieldToShow = column.options.find(nameValue => nameValue.value === field);
-          }
-          return fieldToShow.name;
+    let result: string;
+    result = this.getPersonFieldFromDataRow(personData, column, name);    
+    return result;
+  }
+
+  getPersonFieldFromDataRow(personData: ParticipantData, column: Filter, name: string): string {
+    if (!personData || !personData.data || !name) {
+      return null;
+    }
+    let currentKey = Object.keys(personData.data).find(key => key === name);
+    let field = personData.data[currentKey];
+    if (field) {
+      if (column.options && column.options[0] && column.options[0].name) {
+        let fieldToShow = null;
+        if (column.additionalType === Filter.ACTIVITY_STAFF_TYPE) {
+          fieldToShow = column.options.find(nameValue => nameValue.name === field);
+        } else {
+          fieldToShow = column.options.find(nameValue => nameValue.value === field);
         }
-        return field;
+        return fieldToShow.name;
       }
+      return field;
     }
-    return null;
+    return "";    
+  }
+
+  getPersonFieldForMultipleRows(personDatas: ParticipantData[], column: Filter): string {
+    let name: string
+    if (column && column.participantColumn) {
+      name = column.participantColumn.name;
+    }
+    let result: string;
+    for (let personData of personDatas) {
+      result = this.getPersonFieldFromDataRow(personData, column, name);
+      if (result) {
+        break;
+      }
+    } 
```

Comment:
I think Arrays `find` method would be much more concise instead of for loop.

Reply:
I think this one is more efficient in this case, because we would need map as well and calling the method two times if we used find